server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -672,10 +701,10 @@ private static boolean hasGaps(CachingTokenFilter stream) throws IOException {
             }
             return new PagedBytesIndexFieldData.Builder(fielddataMinFrequency, fielddataMaxFrequency, fielddataMinSegmentSize);
         }
-
     }
 
     private int positionIncrementGap;
+    private Integer lengthLimit;
```

Comment:
Is it necessary to have this property also on the `TextFieldMapper` ? Given that `VARCHAR(lengthLimit)` is mostly introduced for compatibility with SQL/PostgreSQL I think supporting it also for analyzed columns is not really necessary. If somebody adds an analyzer to a column, then it's already CrateDB dialect specific.

Reply:
hm shall we then forbid the declaration of the length limit on analyzed columns in the crate analyzer?

Reply:
Probably makes sense.

One other thing about analyzed columns is that they're tokenized, so whether the length would apply to the individual tokens or the raw input would also be a bit vague.